app/routes/adapter.$name/route.tsx

```diff
@@ -41,23 +41,16 @@ export default function Index() {
     ];
 
     const filteredAdapters = adapters.filter((a) => a.name === name);
-    if (filteredAdapters.length > 0) {
-    }
 
     const adapter = filteredAdapters.length > 0 ? filteredAdapters[0] : null;
 
-    const displayName = (adapter: string) => {
-        if (adapter && adapter.name) {
-            return adapter.name.split('@')[0];
-        }
-        return 'No adapter found';
-    };
+    const displayName = adapter?.name.split('@')[0];
```

Comment:
Veldig bra! 

For å svare på ditt spørsmål, InternalPageHeader er ikke superfan av at typen kan være string | undefined, det er fordi InternalPageHeader forventer at tittelen alltid er en string.

Det kan vi fikse på ved returnere "" når adapter?.name.split('@')[0] er undefined slik: 
`const displayName = adapter?.name.split('@')[0] || '';`